src/Control/Effect/Error.hs

```diff
@@ -1,4 +1,15 @@
 {-# LANGUAGE DeriveFunctor, ExistentialQuantification, FlexibleContexts, StandaloneDeriving #-}
+
+{- | An effect modelling failure with a descriptive error message.
```

Comment:
1. I think we shouldn’t say “descriptive error message” because that’s more like `Fail`; `Error`’s schtick is being polymorphic.

2. We should probably mention that the errors are catchable.